tests/lxd/test_lxd_image_vault.cpp

```diff
@@ -410,25 +435,149 @@ TEST_F(LXDImageVault, has_record_for_returns_expected_values)
+                             mpt::MockLogger::make_cstring_matcher(StrEq("No image update for \'bionic\'."))));
 
     image_vault.update_images(mp::FetchType::ImageOnly, stub_prepare, stub_monitor);
+
+    EXPECT_TRUE(refresh_requested);
+}
+
+TEST_F(LXDImageVault, image_update_source_delete_requested_on_expiration)
+{
+    bool delete_requested{false};
+
+    ON_CALL(*mock_network_access_manager.get(), createRequest(_, _, _))
+        .WillByDefault([&delete_requested](auto, auto request, auto) {
+            auto op = request.attribute(QNetworkRequest::CustomVerbAttribute).toString();
+            auto url = request.url().toString();
+
+            if (op == "GET" && url.contains("1.0/images"))
+            {
+                return new mpt::MockLocalSocketReply(mpt::image_info_update_source_info);
+            }
+            else if (op == "DELETE" &&
+                     url.contains("1.0/images/e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"))
+            {
+                delete_requested = true;
+                return new mpt::MockLocalSocketReply(mpt::image_delete_task_data);
+            }
+
+            return new mpt::MockLocalSocketReply(mpt::not_found_data, QNetworkReply::ContentNotFoundError);
+        });
+
+    mp::LXDVMImageVault image_vault{hosts, mock_network_access_manager.get(), base_url, mp::days{0}};
+
+    EXPECT_CALL(*logger,
+                log(Eq(mpl::Level::info), mpt::MockLogger::make_cstring_matcher(StrEq("lxd image vault")),
+                    mpt::MockLogger::make_cstring_matcher(StrEq("Source image \'bionic\' is expired. Removing it…"))));
+
+    image_vault.prune_expired_images();
+
+    EXPECT_TRUE(delete_requested);
+}
+
+TEST_F(LXDImageVault, image_hash_delete_not_requested_on_expiration)
+{
+    bool delete_requested{false};
+
+    ON_CALL(*mock_network_access_manager.get(), createRequest(_, _, _))
+        .WillByDefault([&delete_requested](auto, auto request, auto) {
+            auto op = request.attribute(QNetworkRequest::CustomVerbAttribute).toString();
+            auto url = request.url().toString();
+
+            if (op == "GET" && url.contains("1.0/images"))
+            {
+                return new mpt::MockLocalSocketReply(mpt::image_info_data);
+            }
+            else if (op == "DELETE" &&
+                     url.contains("1.0/images/e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"))
+            {
+                delete_requested = true;
+                return new mpt::MockLocalSocketReply(mpt::image_delete_task_data);
+            }
+
+            return new mpt::MockLocalSocketReply(mpt::not_found_data, QNetworkReply::ContentNotFoundError);
+        });
+
+    mp::LXDVMImageVault image_vault{hosts, mock_network_access_manager.get(), base_url, mp::days{0}};
+
+    image_vault.prune_expired_images();
+
+    // Should not request delete since image was stored as hash and not alias
+    EXPECT_FALSE(delete_requested);
```

Comment:
Should it not? The difference was for updating (that there's no way to update a fingerprint-queried image), not in pruning?

Reply:
As mentioned above, I _believe_ the default is never to expire hash based images.  If this is not correct, I'll fix here and above.

Reply:
I don't believe that was the desire. See above.

Reply:
Ack, I was wrong about expiring hash based images.  Will fix.

Reply:
Done